lib/autoproj/daemon/github_watcher.rb

```diff
@@ -1,189 +1,224 @@
+                end
             end
 
-            # Checks whether the given 'head' has a PR open to a watched repository
-            #
-            # @param [Repository] repo A wacthed repository definition
-            # @param [String] head The head definition in the format user:branch
-            # @return [Boolean] Whether there's a PR open or not
-            def pullrequest_open?(repo, head)
-                @client.pull_requests("#{repo.owner}/#{repo.name}",
-                                      base: repo.branch).any? do |pr|
-                    pr['head']['label'] == head
+            # @param [Array<Github::PushEvent, Github::PullRequestEvent>] events
+            # @return [(Array<Github::PushEvent>, Array<Github::PullRequestEvent>)]
+            def partition_events_by_type(events)
+                push_events = events.select { |event| event.kind_of? Github::PushEvent }
+                pull_request_events = events.select do |event|
+                    event.kind_of? Github::PullRequestEvent
                 end
+                [push_events, pull_request_events]
             end
 
-            REFS_HEADS_RANGE = ('refs/heads/'.length..-1).freeze
+            # @param [Array<Github::PushEvent, Github::PullRequestEvent>] events
+            # @return [Hash]
+            def partition_events_by_repo_name(events)
+                partitioned = {}
+                events.each do |event|
+                    _, name = event_owner_and_name(event)
+                    partitioned[name] ||= []
+                    partitioned[name] << event
+                end
+                partitioned
+            end
 
-            # Handles (by calling all corresponding hooks) a given set
-            # of events.
-            #
-            # @param [Repository] repo A wacthed repository definition
-            # @param [Array<Hash>] events Hash with the events returned by the API
-            # @return [Integer] The number of handled events
-            #
-            # rubocop: disable Metrics/AbcSize
-            def handle_events(repo, events)
-                filter_events(repo, events).each do |event|
-                    if event['type'] == 'PullRequestEvent'
-                        @pullrequest_hooks.each do |hook|
-                            hook.call("#{repo.owner}/#{repo.name}",
-                                      branch: repo.branch,
-                                      number: event['payload']['number'])
-                        end
-                    else
-                        @push_hooks.each do |hook|
-                            hook.call("#{repo.owner}/#{repo.name}",
-                                      branch: event['payload']['ref'][REFS_HEADS_RANGE])
-                        end
+            # @param [Github::PushEvent] push_event
+            # @return [void]
+            def call_push_hooks(push_event, **options)
+                @push_hooks.each { |hook| hook.call(push_event, options) }
```

Comment:
Use `**options` to forward keyword arguments. This will work until now, but is deprecated (deprecation notices will appear in 2.7, 3.0 will probably remove the possibility)